src/services/rfqm_service.ts

```diff
@@ -296,4 +339,87 @@ export class RfqmService {
             orderHash,
         };
     }
+
+    public async submitMetaTransactionSignedQuoteAsync(
+        params: MetaTransactionSubmitRfqmSignedQuoteParams,
+    ): Promise<MetaTransactionSubmitRfqmSignedQuoteResponse> {
+        const metaTransactionHash = params.metaTransaction.getHash();
+
+        // check that the firm quote is recognized as a previously returned quote
+        const quote = await this.dbUtils.findQuoteByMetaTransactionHashAsync(metaTransactionHash);
+        if (quote === undefined) {
+            RFQM_SIGNED_QUOTE_NOT_FOUND.inc();
+            throw new NotFoundError('quote not found');
+        }
+
+        // validate that the expiration window is long enough to fill quote
+        const currentTimeMs = new Date().getTime();
+        if (
+            !params.metaTransaction.expirationTimeSeconds
+                .times(ONE_SECOND_MS)
+                .isGreaterThan(currentTimeMs - RFQM_MINIMUM_EXPIRY_DURATION_MS)
+        ) {
+            RFQM_SIGNED_QUOTE_EXPIRY_TOO_SOON.inc();
+            throw new InternalServerError(`metaTransaction will expire too soon`);
+        }
+
+        // validate that the firm quote is fillable using the origin registry address (this address is assumed to hold ETH)
+        try {
+            await this._blockchainUtils.validateMetaTransactionOrThrowAsync(
```

Comment:
Do we need to update `validateMetaTransactionOrThrowAsync` to make sure that the full taker amount was filled (as a fix for not having `fillOrKillRfqOrder`)

Reply:
already updated